src/object_file/relocation.rs

```diff
@@ -0,0 +1,111 @@
+use std::marker::PhantomData;
+use std::iter::FusedIterator;
+
+use llvm_sys::object::{LLVMRelocationIteratorRef, LLVMSectionIteratorRef};
+
+use crate::support::LLVMString;
+use super::{Section, Symbol};
+
+#[derive(Debug)]
+pub struct Relocation<'a> {
+    relocation: LLVMRelocationIteratorRef,
+    _marker: PhantomData<&'a Section<'a>>,
+}
+impl<'a> Relocation<'a> {
+    pub(super) fn new(relocation: LLVMRelocationIteratorRef) -> Self {
+        Self {
+            relocation,
+            _marker: PhantomData,
+        }
+    }
+
+    pub fn offset(&self) -> u64 {
+        use llvm_sys::object::LLVMGetRelocationOffset;
+
+        unsafe {
+            LLVMGetRelocationOffset(self.relocation)
+        }
+    }
+
+    pub fn get_type(&self) -> (u64, LLVMString) {
+        use llvm_sys::object::LLVMGetRelocationType;
+        use llvm_sys::object::LLVMGetRelocationTypeName;
+
+        let type_id = unsafe {
+            LLVMGetRelocationType(self.relocation)
+        };
+        let type_name_ptr = unsafe {
+            LLVMGetRelocationTypeName(self.relocation)
+        };
+
+        (type_id, LLVMString::new(type_name_ptr))
+    }
+    
+    pub fn value(&self) -> LLVMString {
+        use llvm_sys::object::LLVMGetRelocationValueString;
+
+        let value_ptr = unsafe {
+            LLVMGetRelocationValueString(self.relocation)
+        };
+
+        LLVMString::new(value_ptr)
+    }
+
+    pub fn symbol(&self) -> Symbol<'a> {
+        use llvm_sys::object::LLVMGetRelocationSymbol;
+
+        let symbol = unsafe {
+            LLVMGetRelocationSymbol(self.relocation)
+        };
+        Symbol::new(symbol)
+    }
+}
+
+#[derive(Debug)]
+pub struct RelocationIterator<'a> {
+    iter: LLVMRelocationIteratorRef,
+    section: LLVMSectionIteratorRef,
+    _marker: PhantomData<&'a Section<'a>>,
```

Comment:
This doesn't look right to me. Since `RelocationIterator` seems to be derived from a `Section`, shouldn't it have `section: &'s Section` as a field rather than the current `section` and `_marker`?